README.md

```diff
@@ -39,10 +48,58 @@ Or install it yourself as:
     $ gem install ar2dto
 
 ## Usage
-TBD
+
+In the following sections, we explain the API provided by the gem and some basic usage.
 
 ### Setting up your models
-TBD
+
+To use the gem, you need to add `has_dto` to your ActiveRecord models. For example:
+```ruby
+class User < ApplicationRecord
+  has_dto
+end
+```
+
+This will dynamically create a class called `UserDTO` and will add two methods to your ActiveRecord model: `#to_dto` and `.to_dto`.
+
+### DTO class
+
+This class is dynamically created based on your models that declare `has_dto`. The goal of these classes is to be data-only. They don't have access to the DB, business logic, or calculate things on the fly. They just store the data and provide them in a read-only fashion through plain simple methods.
+
+In addition to that, and optionally, you can have these objects be compliant with the [ActiveModel API](https://github.com/rails/rails/blob/main/activemodel/lib/active_model/lint.rb) you can do that by configuring it globally with:
+
+```ruby
+# config/initializers/ar2dto.rb
+
+AR2DTO.setup do |config|
+  config.active_model_compliance = true
+end
+```
+
+With this, it'll be even easier to interchange in your project ActiveRecord models for DTOs, because other parts of Rails and other gems will continue to work (e.g. Rails route helpers).
```

Comment:
```suggestion
With this, it'll be even easier to interchange ActiveRecord models for DTOs, because other parts of Rails and other gems will continue to work (e.g. Rails route helpers).
```